### DISCUSSION TITLE
How to create a line with only one end draggable?

### DISCUSSION THREAD

**Discussion by abdussami-tayyab** [fabricjs/fabric.js#8805] [Q&A] [answered]
I have a `fabric.Line` object with only two ends i.e. can only scale in one direction. What I want to do now is to be able to move the line from one end in any direction I want while it's other end stays the same place, just like in the example [here](https://imgur.com/a/PEKDan0).

How can I achieve that? I searched online and couldn't have much success. One way is drawing 2 circles on each end and following it whenever it is moved, but that does not seem a very elegant solution.

**Comment by ShaMan123** [comment, marked as answer]
Use controls, that is exactly for they are for.
https://github.com/fabricjs/fabric.js/blob/master/src/controls/polyControl.ts
https://codesandbox.io/s/custom-control-poly-fix-flip-95y2im
